### PR TITLE
Gives abductors three new equipment shop purchase options, splits up abductor_gear.dm

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

Three new items have been added to the abductor credit store. One carries concrete utility, with the other two being useful for pranking/mind games.

The first is an alien hypertool with toolsets for either hacking or surgery equipment. Toggle by right-clicking the item in your hand!

product demonstration

This can be an equipment upgrade, or replacement, for either member of the team. It costs 2 research credits to purchase.

Also on the list of new stuff to buy (and stuff shown in the gif above) -- You can spend extra points on abductorized versions of existing posters. They can be used as decoration, or to troll and assert dominance over the crew.

You can also buy a cow. It comes with some wheat.

Now, let's talk about all of the impulsive code changes I made while making this PR:
- abduction_gear.dm has been split up into abductor_clothing.dm, abductor_items.dm, and abductor_structures.dm because the file was getting huge and I didn't want to make it bigger.
- There is now a runtime prevention check for trying to probe non-humanoid subjects (like a cow).
- The abductor shop now uses defines for its categories, and uses lists for build paths similarly to cargo crates.

Lastly, to facilitate the hanging of posters on the abductor ship, posters can now be placed on indestructible walls. The code associated with hanging posters (/turf/closed/wall/proc/place_poster) has been moved to the /turf/closed level.

The scope of this PR kind of spiraled out of control with the file splitting and the change to indestructible walls, I'm hoping it isn't too much.

As for the spritework, most of it was done through the cannibalization and recycling of other, existing sprites. Things may not look good (I'm a coder not a spriter!) and I am open to interjections on what can be improved.
## Why It's Good For The Game

The hypertool allows for both the scientist and agent to replace their lost tools (with the same toolspeed), as well as upgrade their current toolset if they have the spare points. Many use cases, all for the same 2 credit package.

The posters, while providing no mechanical advantage, are a nice decoration (for an otherwise cramped, totally undecoratable ship) or way of rubbing your success in the noses of the crew.

The cow is a dual-purpose psychological support/hindrance tool. He can keep you from getting lonely on the mothership, or be beamed down to the crew to annoy them.

Each of these are luxury items that give thriving abductor teams something to spend their excess credits on.
## Changelog
:cl:
add: The abductor equipment vendor now offers a hacking/medical hypertool, at a price of 2 research points.
add: The abductor equipment vendor now sells decorative abductor posters, at a price of 1 research point.
add: The abductor equipment vendor now sells a cow, at the price of 1 research point.
fix: Fixes a runtime when trying to probe non-human mobs.
image: Some abductorized versions of existing wall posters and an alien hypertool.
code: The abductor shop listings now use lists of items, rather than single build paths.
code: The abductor_gear.dm file has been split up into a few different files.
code: Posters can now be hung on indestructible walls.
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
ayy lmao

**Comment by TheVekter** [comment]
I'm a big fan  of the cow, that's a very funny meme.

**Comment by necromanceranne** [comment]
This is probably more complicated than necessary but I would love the idea of the posters looking different to normal crew compared to abduction victims.

Ala They Live. Very much more complicated than it is worth though.

**Review by Wallemations — APPROVED** [review]

**Comment by Serxule** [comment]
make the posters unable to be torn down, so if ayys go troll route the posters cant just be instantly ripped

**Comment by Rhials** [comment]
> make the posters unable to be torn down, so if ayys go troll route the posters cant just be instantly ripped

I actually had this same idea while working on this. I might go ahead with it honestly.

**Comment by Sealed101** [comment]
should crew wirecutters be able to remove these?

**Comment by YakumoChen** [comment]
Can you make the reward vendor display the cow with an excessively scientific  name?

Maybe give it a 20% chance to spawn a wisdom cow instead

File: code/game/turfs/closed/_closed.dm [hunk 1]
```diff
@@ -38,6 +38,12 @@
 /turf/closed/indestructible/singularity_act()
 	return
 
+/turf/closed/indestructible/attackby(obj/item/attacking_item, mob/user, params)
+	..()
+
+	if(istype(attacking_item, /obj/item/poster) && Adjacent(user))
+		return place_poster(attacking_item, user)
```
Comment (san7890) [inline code/game/turfs/closed/_closed.dm:45]: should we `return ..()` instead or is the rest of the attackby code important for reasons beyond my comprehension

File: code/modules/antagonists/abductor/abductor_structures.dm [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		to_chat(user, span_notice("You start disassembling [src]..."))
+		I.play_tool_sound(src)
+		if(I.use_tool(src, user, 30))
+			playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
+			for(var/i in 0 to framestackamount)
+				new framestack(get_turf(src))
+			qdel(src)
+			return
+	if(istype(I, /obj/item/stack/sheet/mineral/abductor))
+		var/obj/item/stack/sheet/P = I
```
Comment (san7890) [inline code/modules/antagonists/abductor/abductor_structures.dm:29]: no new single letter variables please, what the fuck is a `P`

File: code/modules/antagonists/abductor/abductor_structures.dm [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+			to_chat(user, span_warning("You need one alien alloy sheet to do this!"))
+			return
+		to_chat(user, span_notice("You start adding [P] to [src]..."))
+		if(do_after(user, 50, target = src))
+			P.use(1)
+			new /obj/structure/table/abductor(src.loc)
+			qdel(src)
+		return
+	if(istype(I, /obj/item/stack/sheet/mineral/silver))
+		var/obj/item/stack/sheet/P = I
```
Comment (san7890) [inline code/modules/antagonists/abductor/abductor_structures.dm:40]: ditto on yeeting `P`

File: code/modules/antagonists/abductor/abductor_structures.dm [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	buildstack = /obj/item/stack/sheet/mineral/silver
+	framestack = /obj/item/stack/sheet/mineral/abductor
+	buildstackamount = 1
+	framestackamount = 1
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "bed"
+	can_buckle = TRUE
+	buckle_lying = 90
+	/// Amount to inject per second
+	var/inject_am = 0.5
```
Comment (san7890) [inline code/modules/antagonists/abductor/abductor_structures.dm:78]:
```suggestion
	var/inject_amount = 0.5
```

no need to be shy on words, we've them aplenty

File: code/modules/antagonists/abductor/abductor_structures.dm [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+/obj/structure/table/optable/abductor/proc/on_entered(datum/source, atom/movable/AM)
+	SIGNAL_HANDLER
+	if(iscarbon(AM))
+		START_PROCESSING(SSobj, src)
+		to_chat(AM, span_danger("You feel a series of tiny pricks!"))
+
+/obj/structure/table/optable/abductor/process(seconds_per_tick)
+	. = PROCESS_KILL
+	for(var/mob/living/carbon/C in get_turf(src))
```
Comment (san7890) [inline code/modules/antagonists/abductor/abductor_structures.dm:97]: again just yeet this single letter variable

File: code/modules/antagonists/abductor/abductor_structures.dm [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+		. = TRUE
+		for(var/chemical in injected_reagents)
+			if(C.reagents.get_reagent_amount(chemical) < inject_am * seconds_per_tick)
+				C.reagents.add_reagent(chemical, inject_am * seconds_per_tick)
+
+/obj/structure/table/optable/abductor/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
```
Comment (san7890) [inline code/modules/antagonists/abductor/abductor_structures.dm:105]:
```suggestion
	return ..()
```

File: code/modules/antagonists/abductor/abductor_structures.dm [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+		START_PROCESSING(SSobj, src)
+		to_chat(AM, span_danger("You feel a series of tiny pricks!"))
+
+/obj/structure/table/optable/abductor/process(seconds_per_tick)
+	. = PROCESS_KILL
+	for(var/mob/living/carbon/C in get_turf(src))
+		. = TRUE
+		for(var/chemical in injected_reagents)
+			if(C.reagents.get_reagent_amount(chemical) < inject_am * seconds_per_tick)
+				C.reagents.add_reagent(chemical, inject_am * seconds_per_tick)
```
Comment (san7890) [inline code/modules/antagonists/abductor/abductor_structures.dm:101]:
```suggestion
				C.reagents.add_reagent(chemical, inject_am * seconds_per_tick)
				
	return .
```

it's probably okay to have the funny dot operator value for runtime safety in reagent code but you must pay the dot operator toll (explicity goodness)

File: code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+		/obj/item/abductor,
+		/obj/item/melee/baton,
+		/obj/item/gun/energy,
+		/obj/item/restraints/handcuffs
+		)
+	var/mode = VEST_STEALTH
+	var/stealth_active = FALSE
+	/// Cooldown in seconds
+	var/combat_cooldown = 20
+	var/datum/icon_snapshot/disguise
```
Comment (san7890) [inline code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm:36]: would you mind dmdoccing all of these thanks

File: code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+	inhand_icon_state = "armor"
+	blood_overlay_type = "armor"
+	armor_type = /datum/armor/abductor_vest
+	actions_types = list(/datum/action/item_action/hands_free/activate)
+	allowed = list(
+		/obj/item/abductor,
+		/obj/item/melee/baton,
+		/obj/item/gun/energy,
+		/obj/item/restraints/handcuffs
+		)
```
Comment (san7890) [inline code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm:31]:
```suggestion
		/obj/item/restraints/handcuffs,
	)
```

better multilining

File: code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+			DeactivateStealth()
+			set_armor(/datum/armor/abductor_combat)
+			icon_state = "vest_combat"
+		if(VEST_COMBAT)// TO STEALTH
+			mode = VEST_STEALTH
+			set_armor(/datum/armor/abductor_vest)
+			icon_state = "vest_stealth"
+	if(ishuman(loc))
+		var/mob/living/carbon/human/H = loc
+		H.update_worn_oversuit()
```
Comment (san7890) [inline code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm:79]:
```suggestion
		var/mob/living/carbon/human/target = loc
		target.update_worn_oversuit()
```

File: code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+
+/obj/item/clothing/suit/armor/abductor/vest/proc/SetDisguise(datum/icon_snapshot/entry)
+	disguise = entry
+
+/obj/item/clothing/suit/armor/abductor/vest/proc/ActivateStealth()
+	if(disguise == null)
+		return
+	stealth_active = TRUE
+	if(ishuman(loc))
+		var/mob/living/carbon/human/M = loc
```
Comment (san7890) [inline code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm:94]: there's a bunch of these that should be exterminated through this file

File: code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+	if(combat_cooldown >= initial(combat_cooldown))
+		STOP_PROCESSING(SSobj, src)
+
+/obj/item/clothing/suit/armor/abductor/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	for(var/obj/machinery/abductor/console/C in GLOB.machines)
+		if(C.vest == src)
+			C.vest = null
+			break
+	. = ..()
```
Comment (san7890) [inline code/modules/antagonists/abductor/equipment/gear/abductor_clothing.dm:156]:
```suggestion
	return ..()
```

File: code/modules/antagonists/abductor/equipment/gear/abductor_items.dm [hunk 13]
```diff
@@ -768,152 +612,103 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+
+/obj/item/abductor/alien_omnitool/proc/set_toolset(mob/user)
+	if(active_toolset == TOOLSET_MEDICAL)
+		tool_list =	list("Crowbar" = image(icon = 'icons/obj/abductor.dmi', icon_state = "crowbar"),
+			"Multitool" = image(icon = 'icons/obj/abductor.dmi', icon_state = "multitool"),
+			"Screwdriver" = image(icon = 'icons/obj/abductor.dmi', icon_state = "screwdriver_a"),
+			"Wirecutters" = image(icon = 'icons/obj/abductor.dmi', icon_state = "cutters"),
+			"Wrench" = image(icon = 'icons/obj/abductor.dmi', icon_state = "wrench"),
+			"Welding Tool" = image(icon = 'icons/obj/abductor.dmi', icon_state = "welder"),
+		)
```
Comment (san7890) [inline code/modules/antagonists/abductor/equipment/gear/abductor_items.dm:697]:
```suggestion
		tool_list =list(
			"Crowbar" = image(icon = 'icons/obj/abductor.dmi', icon_state = "crowbar"),
			"Multitool" = image(icon = 'icons/obj/abductor.dmi', icon_state = "multitool"),
			"Screwdriver" = image(icon = 'icons/obj/abductor.dmi', icon_state = "screwdriver_a"),
			"Wirecutters" = image(icon = 'icons/obj/abductor.dmi', icon_state = "cutters"),
			"Wrench" = image(icon = 'icons/obj/abductor.dmi', icon_state = "wrench"),
			"Welding Tool" = image(icon = 'icons/obj/abductor.dmi', icon_state = "welder"),
		)
```

multiline it right, may as well

File: code/modules/antagonists/abductor/equipment/gear/abductor_items.dm [hunk 14]
```diff
@@ -768,152 +612,103 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	else
+		tool_list = list(
+			"Retractor" = image(icon = 'icons/obj/abductor.dmi', icon_state = "retractor"),
+			"Hemostat" = image(icon = 'icons/obj/abductor.dmi', icon_state = "hemostat"),
+			"Cautery" = image(icon = 'icons/obj/abductor.dmi', icon_state = "cautery"),
+			"Drill" = image(icon = 'icons/obj/abductor.dmi', icon_state = "drill"),
+			"Scalpel" = image(icon = 'icons/obj/abductor.dmi', icon_state = "scalpel"),
+			"Saw" = image(icon = 'icons/obj/abductor.dmi', icon_state = "saw"),
+			"Bonesetter" = image(icon = 'icons/obj/medical/surgery_tools.dmi', icon_state = "bonesetter"),
+			"Blood Filter" = image(icon = 'icons/obj/medical/surgery_tools.dmi', icon_state = "bloodfilter")
```
Comment (san7890) [inline code/modules/antagonists/abductor/equipment/gear/abductor_items.dm:710]:
```suggestion
			"Blood Filter" = image(icon = 'icons/obj/medical/surgery_tools.dmi', icon_state = "bloodfilter"),
```

File: code/modules/antagonists/abductor/equipment/gear/abductor_posters.dm [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+	icon = 'icons/obj/abductor_posters.dmi'
+	poster_item_name = "abductor poster"
+	poster_item_desc = "A sheet of holofiber resin, with a nanospike perforation on the back end for maximum adhesion."
+	poster_item_icon_state = "rolled_abductor"
+
+/obj/structure/sign/poster/abductor/tear_poster(mob/user)
+	if(!isabductor(user))
+		balloon_alert(user, "it won't budge!")
+		return
+	..()
```
Comment (san7890) [inline code/modules/antagonists/abductor/equipment/gear/abductor_posters.dm:18]: why not

```suggestion
	return ..()
```

**Review by san7890 — CHANGES_REQUESTED** [review]
i wasn't sure which was your code and which was reorganized so i just marked it all down. if you don't have time for the really nitty stuff that you didn't add lmk, but they all seem trivial enough to my eyes

File: code/modules/antagonists/abductor/abductor_structures.dm (hunk 2 above)
Comment (san7890) [reply code/modules/antagonists/abductor/abductor_structures.dm:29]: i later realize that this likely isn't a "new" thing, but should still be eradicated

File: code/game/turfs/closed/_closed.dm (hunk 1 above)
Comment (Rhials) [reply code/game/turfs/closed/_closed.dm:45]: Do you mean putting return ..() after the poster check? Also I don't think the earlier attackby code is important in this context.

Comment (san7890) [reply code/game/turfs/closed/_closed.dm:45]: yes i meant after. sorry for getting back to you so late

**Pull request merged**